src/pocketmine/plugin/PluginManager.php

```diff
@@ -267,6 +267,16 @@ public function loadPlugins(string $directory, array $newLoaders = null){
 						continue;
 					}
 
+					$pluginPhpVersions = $description->getCompatiblePhpVersions();
+					preg_match("#^\d+(\.\d+)*#", PHP_VERSION, $matches);
```

Comment:
You could use the `PHP_MAJOR_VERSION`, `PHP_MINOR_VERSION` and `PHP_RELEASE_VERSION` constants for this instead of regexing. Also, consider making a function for version comparison so that it can be unit-tested.

Reply:
Thanks for the feedback, this should be resolved now, I added `Utils::isVersionCompatible()` for unit testing.